src/Fields/field.jl

```diff
@@ -158,21 +251,30 @@ iterate(f::Field, state=1) = iterate(f.data, state)
 @inline znode(::Type{Cell}, k, grid) = @inbounds grid.zC[k]
 @inline znode(::Type{Face}, k, grid) = @inbounds grid.zF[k]
 
-@inline xnode(i, ϕ::Field{X, Y, Z}) where {X, Y, Z} = xnode(X, i, ϕ.grid)
-@inline ynode(j, ϕ::Field{X, Y, Z}) where {X, Y, Z} = ynode(Y, j, ϕ.grid)
-@inline znode(k, ϕ::Field{X, Y, Z}) where {X, Y, Z} = znode(Z, k, ϕ.grid)
+@inline xnode(i, ψ::Field{X, Y, Z}) where {X, Y, Z} = xnode(X, i, ψ.grid)
+@inline ynode(j, ψ::Field{X, Y, Z}) where {X, Y, Z} = ynode(Y, j, ψ.grid)
+@inline znode(k, ψ::Field{X, Y, Z}) where {X, Y, Z} = znode(Z, k, ψ.grid)
+
+# Dispatch insanity
+xnodes(::Type{Cell}, topo, grid) = reshape(grid.xC, grid.Nx, 1, 1)
+ynodes(::Type{Cell}, topo, grid) = reshape(grid.yC, 1, grid.Ny, 1)
+znodes(::Type{Cell}, topo, grid) = reshape(grid.zC, 1, 1, grid.Nz)
+
+xnodes(::Type{Face}, topo, grid) = reshape(grid.xF[1:end-1], grid.Nx, 1, 1)
+ynodes(::Type{Face}, topo, grid) = reshape(grid.yF[1:end-1], 1, grid.Ny, 1)
+znodes(::Type{Face}, topo, grid) = reshape(grid.zF[1:end-1], 1, 1, grid.Nz)
 
-xnodes(ϕ::AbstractField) = reshape(ϕ.grid.xC, ϕ.grid.Nx, 1, 1)
-ynodes(ϕ::AbstractField) = reshape(ϕ.grid.yC, 1, ϕ.grid.Ny, 1)
-znodes(ϕ::AbstractField) = reshape(ϕ.grid.zC, 1, 1, ϕ.grid.Nz)
+xnodes(::Type{Face}, ::Bounded, grid) = reshape(grid.xF, grid.Nx+1, 1, 1)
+ynodes(::Type{Face}, ::Bounded, grid) = reshape(grid.yF, 1, grid.Ny+1, 1)
+znodes(::Type{Face}, ::Bounded, grid) = reshape(grid.zF, 1, 1, grid.Nz+1)
 
-xnodes(ϕ::Field{Face})                    = reshape(ϕ.grid.xF[1:end-1], ϕ.grid.Nx, 1, 1)
-ynodes(ϕ::Field{X, Face}) where X         = reshape(ϕ.grid.yF[1:end-1], 1, ϕ.grid.Ny, 1)
-znodes(ϕ::Field{X, Y, Face}) where {X, Y} = reshape(ϕ.grid.zF[1:end-1], 1, 1, ϕ.grid.Nz)
+xnodes(ψ::AbstractField) = xnodes(x_location(ψ), x_topology(ψ), ψ.grid)
+ynodes(ψ::AbstractField) = ynodes(y_location(ψ), y_topology(ψ), ψ.grid)
+znodes(ψ::AbstractField) = znodes(z_location(ψ), z_topology(ψ), ψ.grid)
 
-nodes(ϕ) = (xnodes(ϕ), ynodes(ϕ), znodes(ϕ))
+nodes(ψ::AbstractField) = (xnodes(ψ), ynodes(ψ), znodes(ψ))
 
-# Niceties
+# Nicites (but what for?)
```

Comment:
Typo but I think you added these! They're used for dispatch in `set!.jl` right?

Guess it's worth actually commenting what niceties are for haha.

Reply:
I will check that.

Reply:
Ok, I moved them to `set!.jl`, where they are used. I had added them prior to the splitting of code into files; I couldn't remember where they ended up.